### PR TITLE
leverage toolchain.linalg in ScaLAPACK easyblock, use copy_file

### PR DESCRIPTION
**Pull request by boegel**
required to get rid of implementation that was deprecated for EasyBuild v3.0

this should not affect existing easyconfigs, it's just implementation details really...

### CODE REVIEW

**Comment by akesandgren** [comment]
Actually MKL should probably be in the dep_grp for scalapack too
There are situations where one have to use the netlib scalapack and mkl together....
there be bugs in mkl scalapack in some cases (at least there used to be)

**Comment by boegel** [comment]
@akesandgren I suggest we add that in later as needed? I have yet to see a ScaLAPACK easyconfig that sit on top of `imkl`, let alone a toolchain that leverages it...

**Comment by boegel** [comment]
@wpoely86 please review?

File: easybuild/easyblocks/s/scalapack.py [hunk 1]
```diff
@@ -56,33 +59,28 @@ def configure_step(self):
...
+        else:
+            copy_file(src, dest)
 
         self.loosever = LooseVersion(self.version)
 
-        # make sure required dependencies are available
-        deps = [("LAPACK", "ACML", "OpenBLAS")]
-        self.log.deprecated("EB_ScaLAPACK.configure_step uses hardcoded list of LAPACK libs", '3.0')
+        # make sure required dependencies are available for BLAS/LAPACK
+        dep_grps = [('ACML', 'LAPACK', 'OpenBLAS')]
```
Comment (wpoely86) [inline easybuild/easyblocks/s/scalapack.py]: didn't you add constants for these?

Comment (boegel) [inline easybuild/easyblocks/s/scalapack.py]: these are the software names, but I guess I can use `Acml.BLAS_MODULE_NAME[0]` & co here, like I do in #1016?

**Comment by boegel** [comment]
@wpoely86: please rereview?

**Comment by boegel** [comment]
tested on all existing ScaLAPACK easyconfigs, good to go, thanks for the review @wpoely86!

**Pull request merged**